docs/src/component-statuses.js

```diff
@@ -0,0 +1,67 @@
+import {Link} from '@primer/components'
+import StatusLabel from '@primer/gatsby-theme-doctocat/src/components/status-label'
+import Table from '@primer/gatsby-theme-doctocat/src/components/table'
+import {graphql, Link as GatsbyLink, useStaticQuery} from 'gatsby'
+import React from 'react'
+
+export function ComponentStatuses() {
+  const data = useStaticQuery(graphql`
+    query ComponentStatuses {
+      allMdx(
+        filter: {frontmatter: {status: {ne: null}}}
+        sort: {fields: frontmatter___title}
+      ) {
+        nodes {
+          slug
+          excerpt(pruneLength: 300)
+          frontmatter {
+            status
+            title
+          }
+        }
+      }
+    }
+  `)
+
+  const components = data.allMdx.nodes.map((node) => {
+    // get first sentence of the excerpt
+    // search for a period space and capital letter to avoid splitting at e.g.
+    const match = node.excerpt
+      .replace(/\n/g, ' ')
+      .match(/(^.+?\.)(\s+[A-Z])/, '.')
```

Comment:
This is a bit sketchy. Would we want to have descriptions defined separately? I thought about updating the `docs.rake` script to add a frontmatter description to each of the md files but that would have taken me a lot longer than doing it this way 😅 and have similar sanitisation issues as this... thoughts welcome

Reply:
Should this PR also include copy changes so the descriptions have the same format? (Use X for Y)

Reply:
Here's a list of how we could implement component descriptions (roughly increasing in "correctness"):

1. Parse the first sentence of page content (this is what your PR currently does)
2. Update the docs setup to use the `description` frontmatter variable. This will render the description slightly larger on the page and set the `meta` description tag (so better for SEO).
3. Move component descriptions into the [component-metadata](https://github.com/primer/component-metadata) repo and reference that metadata with the [componentId](https://primer.style/doctocat/usage/front-matter#component-metadata) frontmatter variable

I'll leave it up to you to decide what feels right to implement in this PR. We can take an incrementally correct approach here ✨

Reply:
Now you've explained to me the intentions of the component-metadata repo that seems like easily the best choice here, but I think this is a good interim solution until all the component data is in that repo because otherwise the descriptions are quite sparse.

Since here and #1656 are doing almost the same thing maybe we could build this page into doctocat based on pages that have `componentId` defined?

Reply:
Sounds good. Let's ship this as-is then collaborate on improving the component-metadata package.